ticketutil/servicenow.py

```diff
@@ -0,0 +1,321 @@
+        self.s.headers.update({'Content-Type': 'application/json',
+                               'Accept': 'application/json'})
+        response = self.s.post(url, data=data)
+        logging.debug('POST request Status Code: {0}'
+                      .format(response.status_code))
+        return self.process_response(response)
+
+    def api_put(self, url, data):
+        """
+        API PUT wrapper
+        Updates table content.
+
+        :param url: url containing REST API query
+        :param text: text to be inserted into table record
+        """
+        self.s.headers.update({'Content-Type': 'application/json',
+                               'Accept': 'application/json'})
+        response = self.s.put(url, data=data)
+        logging.debug('PUT request Status Code: {0}'
+                      .format(response.status_code))
+        return self.process_response(response)
+
+    def get_sys_id(self):
+        """
+        Get sys_id query wrapper
+        TODO: what about security, encodeURIComponent in python?
+
+        :return: sys_id
+        """
+        url = self.rest_url + '?sysparm_query=GOTOnumber%3D'
+        url += self.ticket_id
+        result = self.api_get(url)
+        try:
+            return result[0]['sys_id']
+        except:
+            self.ticket_id = None
+            self.sys_id = None
+            logging.error("Failed to get sys_id")
+
+    def _generate_ticket_url(self):
+        """
+        Generates the ticket URL out of the url, project, and ticket_id.
+
+        :return: ticket_url: The URL of the ticket.
+        """
+        ticket_url = None
+
+        # If we are receiving a ticket_id, set ticket_url.
+        if self.ticket_id:
+            self.sys_id = self.get_sys_id() if not self.sys_id else self.sys_id
+            ticket_url = '{0}/{1}.do?sys_id={2}'.format(self.url, self.table,
+                                                        self.sys_id)
+        return ticket_url
+
+    def create(self,
+               short_description,
+               description,
+               u_category,
+               u_item,
+               **kwargs):
+        """
+        Creates new issue, new record in the ServiceNow table
+
+        :param short_description: short description of the issue
+        :param description: full description of the issue
+        :param u_category: ticket category (Category in WebUI)
+        :param u_item: ticket category item (Item in WebUI)
+
+        :param kwargs: optional fields
+
+        Fields example:
+        'contact_type': 'Email',
+        'u_opened_for': 'PNT',
+        'assigned_to': 'pzubaty',
+        'impact': '2',
+        'urgency': '2',
+        'priority': '2'
+        """
+        fields = {'description': description,
+                  'u_category': u_category,
+                  'u_item': u_item}
+        kwargs.update(fields)
+        self.create_simple(short_description, **kwargs)
+
+    def create_simple(self, short_description, **kwargs):
```

Comment:
As I wrote above, this was the original create method. I left it in case it would be used in a different ServiceNow (with different mandatory fields), but I can remove it.